feature/account/edit/src/test/kotlin/app/k9mail/feature/account/edit/domain/usecase/LoadAccountStateTest.kt

```diff
@@ -0,0 +1,139 @@
+package app.k9mail.feature.account.edit.domain.usecase
+
+import app.k9mail.feature.account.common.data.InMemoryAccountStateRepository
+import app.k9mail.feature.account.common.domain.entity.AccountOptions
+import app.k9mail.feature.account.common.domain.entity.AccountState
+import app.k9mail.feature.account.common.domain.entity.AuthorizationState
+import app.k9mail.feature.account.common.domain.entity.MailConnectionSecurity
+import assertk.assertThat
+import assertk.assertions.isEqualTo
+import com.fsck.k9.mail.AuthType
+import com.fsck.k9.mail.ServerSettings
+import kotlinx.coroutines.test.runTest
+import org.junit.Test
+
+class LoadAccountStateTest {
+
+    @Test
+    fun `should load account state WHEN account in state repository has different UUID`() = runTest {
+        val accountState = AccountState(
+            uuid = "differentUuid",
+            emailAddress = EMAIL_ADDRESS,
+            incomingServerSettings = INCOMING_SERVER_SETTINGS,
+            outgoingServerSettings = OUTGOING_SERVER_SETTINGS,
+            authorizationState = AUTHORIZATION_STATE,
+            options = OPTIONS,
+        )
+
+        val testSubject = LoadAccountState(
+            accountStateLoader = { accountUuid ->
+                accountState.copy(uuid = accountUuid)
+            },
+            accountStateRepository = InMemoryAccountStateRepository(),
+        )
+
+        val result = testSubject.execute(ACCOUNT_UUID)
+
+        assertThat(result).isEqualTo(
+            AccountState(
+                uuid = ACCOUNT_UUID,
+                emailAddress = EMAIL_ADDRESS,
+                incomingServerSettings = INCOMING_SERVER_SETTINGS,
+                outgoingServerSettings = OUTGOING_SERVER_SETTINGS,
+                authorizationState = AUTHORIZATION_STATE,
+                options = OPTIONS,
+            ),
+        )
+    }
+
+    @Test
+    fun `should return account state WHEN account in state repository has same UUID`() = runTest {
+        val accountState = AccountState(
+            uuid = ACCOUNT_UUID,
+            emailAddress = EMAIL_ADDRESS,
+            incomingServerSettings = INCOMING_SERVER_SETTINGS,
+            outgoingServerSettings = OUTGOING_SERVER_SETTINGS,
+            authorizationState = AUTHORIZATION_STATE,
+            options = OPTIONS,
+        )
+
+        val testSubject = LoadAccountState(
+            accountStateLoader = { accountUuid ->
+                accountState.copy(uuid = accountUuid)
```

Comment:
With this test one can't tell where the result is coming from. If we change this line to

```kotlin
fail("AccountStateLoader should not be called in this test")
```

we can be sure it's not coming from `AccountStateLoader`.